src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -623,45 +620,73 @@ ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
         return UCS_OK; /* all sends completed */
     }
 
-    ucs_assert(status == UCS_INPROGRESS);
-    ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
+    if (ep->flags & UCT_DC_MLX5_EP_FLAG_FLUSH_CANCEL) {
```

Comment:
/* Must be after uct_dc_mlx5_iface_flush_dci() so if QP is flushed already we will return OK for the subsequent flush(cancel) operations */

Reply:
done